code/datums/elements/update_icon_blocker.dm

```diff
@@ -0,0 +1,10 @@
+//Prevents calling anything in update_icon() like update_icon_state() or update_overlays()
+
+/datum/element/update_icon_blocker/Attach(atom/target)
+	if(!istype(target))
```

Comment:
I'm really not a fan of this pattern where the arg type may be incorrect and it's checked in the proc. Please keep the arg type correct for what could come in and cast it to another var if you need to istype it. In this case though you dont even need it cast properly